askama_derive/src/generator.rs

```diff
@@ -1053,6 +1074,33 @@ impl<'a> Generator<'a> {
         Ok(size_hint)
     }
 
+    fn handle_embed(
+        &mut self,
+        buf: &mut Buffer,
+        embed: &'a Embed<'_>,
+    ) -> Result<usize, CompileError> {
+        self.flush_ws(embed.ws1);
+        self.write_buf_writable(buf)?;
+        let embed_path = self
+            .input
+            .config
+            .find_template(embed.path, Some(&self.input.path))?;
+        let mut embedded_context =
+            Context::new(self.input.config, &self.input.path, embed.nodes.as_slice())?;
```

Comment:
There's missing errors for doing anything outside block tags. Some of these don't actually do anything (extends for example gets overwritten), but you're allowed to write macro and include nodes here, which seems weird.

Reply:
Yes, I see why this is problematic, but it's consistent. I think this is something, that should be fixed/improved in general.

Another real problem, that I mentioned in the issue (#488) is, that the compiler is not including the file when the `embed` is inside a `block` in an `extend`ed file. This problem however also exists for `include`s and I'm not sure how to approach this.